app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java

```diff
@@ -198,9 +200,30 @@ private void handleSuccess(List<Media> mediaList) {
             imagesAdapter.addAll(mediaList);
             imagesAdapter.notifyDataSetChanged();
             ((SearchActivity)getContext()).viewPagerNotifyDataSetChanged();
+            for (Media m : mediaList) {
+                replaceTitlesWithCaptions(m.getDisplayTitle(), mediaSize++);
+            }
         }
     }
 
+        public void replaceTitlesWithCaptions(String displayTitle, int i) {
+            compositeDisposable.add(mediaClient.getCaptionByFilename("File:"+displayTitle+".jpg")
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                    .subscribe(subscriber -> {
+                        handleLabelforImage(subscriber, i);
+                    }));
+
+        }
+
+        private void handleLabelforImage(String s, int position) {
+            if (!s.trim().equals(getString(R.string.detail_caption_empty))) {
```

Comment:
Is the else intentionally omitted, is "s" supposed to be non-null, if yes please add the corresponding annotation

Reply:
@ashishkumar468 Yes the string obtained is not null. I have added the annotation in the next commit

Reply:
@ashishkumar0207 the else is not omitted we want id the condition is false then replace title with caption hence keep the title as it was. Also I have added the annotation